core/src/HIP/Kokkos_HIP_Instance.hpp

```diff
@@ -70,7 +74,7 @@ class HIPInternal {
  public:
   using size_type = ::Kokkos::HIP::size_type;
 
-  static int m_hipDev;
+  int m_hipDev = -1;
   static int m_maxThreadsPerSM;
 
   static hipDeviceProp_t m_deviceProp;
```

Comment:
Just noting that this is still static.
Not blocking considering that the CUDA backend has the same issue but this is something that needs to be looked into.